### ISSUE TITLE
Updates can hang the system due to IO

### ISSUE THREAD

**Issue by gicmo** [flatpak/flatpak#2322]
System was frozen and hanging because GNOME Software did a background update of flatpak (`flatpak-system-helper`) SDKs/runtimes  (around 2GB in size). @hughsie suggested to limit the IO of the worker (as apparently GNOME software already does). Alex thinks it might be the `sync()` from ostree:
> @alexlarsson but generally such things are from the sync when making the ostree transactions commit

**Comment by matthiasclasen** [comment]
We should get the Endless work on IO priorities upstreamed.

**Comment by mwleeds** [comment]
Certainly https://github.com/flatpak/flatpak/commit/060322b50bcaa68bc39966db2f352878e94013e1 is upstream. I don't know if there's other work that needs upstreaming.

**Comment by mwleeds** [comment]
@gicmo could you mention what flatpak version you have?

**Comment by kirbyfan64** [comment]
If the entire system was frozen, could it have been CPU usage? That's the main performance hit I've personally experienced with automatic updates.

**Comment by matthiasclasen** [comment]
> Certainly [060322b](https://github.com/flatpak/flatpak/commit/060322b50bcaa68bc39966db2f352878e94013e1) is upstream. I don't know if there's other work that needs upstreaming.

I'm not sure about the status of the gnome-software changes

**Comment by matthiasclasen** [comment]
turns out the gnome-software changes are merged as well.

**Comment by matthiasclasen** [comment]
there's more improvements in the pipeline (bindfs).

**Comment by mwleeds** [comment]
Yeah, for the record those are here: https://github.com/endlessm/flatpak/pull/115

**Comment by uajain** [comment]
> Yeah, for the record those are here: [endlessm/flatpak#115](https://github.com/endlessm/flatpak/pull/115)

I have opened a upstream PR : https://github.com/flatpak/flatpak/pull/2342

**Comment by johnseekins** [comment]
Just experienced this issue this morning in Fedora 34:
![image](https://user-images.githubusercontent.com/5665929/128887026-08584d1b-55bb-4dce-bef6-f11a6f762995.png)
